### ISSUE TITLE
Can all the Visual Tokenizer weights released work for the same Infinity-2B model?

### ISSUE THREAD

**Issue by EternalEvan** [FoundationVision/Infinity#15]
Thanks for your excellent work! I noticed that you have released many Visual Tokenizer weights with different codebook size. I wonder if all these tokenizer work well with the Infinity-2B weights? I have tried the recommended `infinity_vae_d32reg.pth` and it performers well. Thanks!

**Comment by JeyesHan** [comment]
@EternalEvan Thanks for your appreciation to Infinity. Infinity-2B weight is trained and therefore works with [infinity_vae_d32reg.pth]. Using vae weights will generate bad images. If you want to try other vae weights, you can fine-tune Infinity-2B with other vae weights we provided. The results will improved very quickly.

**Comment by EternalEvan** [comment]
> @EternalEvan Thanks for your appreciation to Infinity. Infinity-2B weight is trained and therefore works with [infinity_vae_d32reg.pth]. Using other vae weights will generate bad images. If you want to try other vae weights, you can fine-tune Infinity-2B with them. The results will improve very quickly.

ok, I will try tuning Infinity-2B with them. Thanks!

**Comment by RealAntonVoronov** [comment]
Can you explain, what does `_reg` mean in VAE_d32? I see in your table with metrics (and confirm by comparing reconstructions from both VAEs) that d32 without _reg works better. What is the reason behind choosing d32_reg for final model?

**Comment by JeyesHan** [comment]
@RealAntonVoronov We experimentally found that as the vocabulary size increases, VAE relies more on the last few scales. In the model with '_reg', we added some regularizations (to be more specific, adding reconstruction loss to the earlier scales). The '_reg' model shows a slight decrease in reconstruction metrics compared to that one without regularization. However, it reduces the dependence on the last few scales, which is beneficial for generation.

**Comment by znavidi** [comment]
Thanks for your great work and guidance regarding the VAE weights provided. I successfully fine-tuned and inferred the 2B Infinity model based on the [infinity_vae_d32_reg.pth](https://huggingface.co/FoundationVision/Infinity/blob/main/infinity_vae_d32reg.pth) VAE, however, when I try with [infinity_vae_d32.pth](https://huggingface.co/FoundationVision/infinity/blob/main/infinity_vae_d32.pth) VAE with the exact same settings (parameters), it raises the following error when loading the VAE. 

do I need to modify any other input parameters for fine-tuning with other VAE weights?

<img width="906" alt="Image" src="https://github.com/user-attachments/assets/857c5e55-a2db-44d7-8458-9aa34b622902" />

Looking forward to your guidance. Thank you!

**Comment by Joanna-0421** [comment]
@EternalEvan @JeyesHan Hi, have you success for finetune the Infinity-2B with other VAE models? How many iters can get a good result for generation. 
When I finetune the Infinity-2B with other VAE, the accuracy truely improved quickly (bit acc ~80% when 4k iters), but the model can't generate. The generated images are meaningless. I think it maybe because the tokens are not align with text, so the generation process can't be guided by the   prompt. Can you give some suggestions about the parameter settings, such as iters, batchsize, total samples and learning rate.
Thanks!!

**Comment by znavidi** [comment]
@Joanna-0421 could you please clarify what you mean by " I think it maybe because the tokens are not align with text, so the generation process can't be guided by the prompt". From what I understand, shouldn't the fine-tuning help with the alignment between the condition (prompt) encoding and image encoding? thanks
